OrbitQt/orbitmainwindow.cpp

```diff
@@ -640,6 +641,13 @@ void OrbitMainWindow::UpdateCaptureStateDependentWidgets() {
   if (hint_frame_ != nullptr) {
     hint_frame_->setVisible(!has_data);
   }
+
+  // Gray out disabled actions on the capture toolbar.
+  for (QAction* action : ui->capture_toolbar->actions()) {
+    QGraphicsOpacityEffect* effect = new QGraphicsOpacityEffect;
+    effect->setOpacity(action->isEnabled() ? 1 : 0.3);
+    ui->capture_toolbar->widgetForAction(action)->setGraphicsEffect(effect);
```

Comment:
This heap allocation without the corresponding `delete` feels a little bit wrong to me, but I guess this is the qt way and there is no way around it. 

Maybe you can put a small comment explaining that `setGraphicsEffect()` transfers the ownership of effect to the QWidget and it means that effect actually gets deleted at some point.